src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp

```diff
@@ -278,9 +278,8 @@ ASTPtr PostgreSQLReplicationHandler::getCreateNestedTableQuery(StorageMaterializ
 {
     postgres::Connection connection(connection_info);
     pqxx::nontransaction tx(connection.getRef());
-    auto table_structure = std::make_unique<PostgreSQLTableStructure>(fetchPostgreSQLTableStructure(tx, table_name, postgres_schema, true, true, true));
-    if (!table_structure)
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Failed to get PostgreSQL table structure");
+    auto [postgres_table_schema, postgres_table_name] = getSchemaAndTableName(table_name);
+    auto table_structure = std::make_unique<PostgreSQLTableStructure>(fetchPostgreSQLTableStructure(tx, postgres_table_name, postgres_table_schema, true, true, true));
 
     auto table_override = tryGetTableOverride(current_database_name, table_name);
```

Comment:
If result of `tryGetTableOverride` as always `ASTTableOverride`, why do we return `ASTPtr` and then cast it? Not related to this changes, though

Reply:
> If result of tryGetTableOverride as always ASTTableOverride

It must be shared ptr anyway, so then by your suggestion it could be `std::shared_ptr<ASTTableOverride>`, but usually as I see even in such cases ASTPtr is used.